pkg/session/session.go

```diff
@@ -187,42 +188,54 @@ func (sess *Session) Fetch(QUri *frontierV1.QueuedUri, crawlConf *configV1.Confi
 		}
 	}()
 
+	span, ctx := opentracing.StartSpanFromContext(ctx, "session",
+		opentracing.Tag{Key: "session.id", Value: sess.Id})
+	defer span.Finish()
+
 	log.WithField("eid", QUri.ExecutionId).Infof("Start fetch of %v", QUri.Uri)
 	sess.RequestedUrl = QUri
 	sess.CrawlConfig = crawlConf.GetCrawlConfig()
 
-	bConf, err := sess.DbAdapter.GetConfigObject(sess.CrawlConfig.BrowserConfigRef)
+	bConf, err := sess.DbAdapter.GetConfigObject(ctx, sess.CrawlConfig.BrowserConfigRef)
 	if err != nil {
 		return nil, fmt.Errorf("failed to get browser config: %v", err)
 	}
-	sess.BrowserConfig = bConf.GetBrowserConfig()
+	sess.browserConfig = bConf.GetBrowserConfig()
 
-	sess.PolitenessConfig, err = sess.DbAdapter.GetConfigObject(sess.CrawlConfig.PolitenessRef)
+	sess.PolitenessConfig, err = sess.DbAdapter.GetConfigObject(ctx, sess.CrawlConfig.PolitenessRef)
 	if err != nil {
 		return nil, fmt.Errorf("failed to get politeness config: %v", err)
 	}
 
-	maxTotalTime := time.Duration(sess.BrowserConfig.PageLoadTimeoutMs) * time.Millisecond
-	maxIdleTime := time.Duration(sess.BrowserConfig.MaxInactivityTimeMs) * time.Millisecond
+	maxTotalTime := time.Duration(sess.browserConfig.PageLoadTimeoutMs) * time.Millisecond
+	maxIdleTime := time.Duration(sess.browserConfig.MaxInactivityTimeMs) * time.Millisecond
+
+	if scripts, err := sess.loadScripts(ctx); err != nil {
+		return nil, fmt.Errorf("failed to load scripts: %w", err)
+	} else {
+		sess.scripts = scripts
+	}
 
-	allocatorContext, allocatorCancel := chromedp.NewRemoteAllocator(context.Background(), sess.browserWsEndpoint)
+	allocatorContext, allocatorCancel := chromedp.NewRemoteAllocator(ctx, sess.browserWsEndpoint)
```

Comment:
Litt usikker på om du risikerer at Allocator blir avbrutt av at context blir kansellert før man får ryddet opp. Tror det var derfor jeg laget ny context her. Men jeg er usikker. Går god for det hvis du har gjort tilstrekkelig med stresstester.

Reply:
Must use the passed context to get the tracing span. Fixed by not passing through the root context in the harvester component.